### ISSUE TITLE
[Carvel] Timeout occurs when try to see the applications deployed - http://<VM_IP>/#/c/default/ns/tap-install/apps

### ISSUE THREAD

**Issue by cmoulliard** [vmware-tanzu/kubeapps#4542]
## Issue

Timeout occurs when try to see the applications deployed using as URL (e.g.:  http://<VM_IP>/#/c/default/ns/tap-install/apps)

![Screenshot 2022-03-30 at 14 18 38](https://user-images.githubusercontent.com/463790/160833054-92515ef0-f4e3-4f73-87cf-e13030b21b64.png)

```
kc get pkgi -A
NAMESPACE     NAME                       PACKAGE NAME                                   PACKAGE VERSION   DESCRIPTION           AGE
default       my-dashboard               kubernetes-dashboard.halkyonio.io              0.1.0             Reconcile succeeded   44h
tap-install   accelerator                accelerator.apps.tanzu.vmware.com              1.0.2             Reconcile succeeded   5d22h
tap-install   appliveview                run.appliveview.tanzu.vmware.com               1.0.2             Reconcile succeeded   5d22h
tap-install   appliveview-conventions    build.appliveview.tanzu.vmware.com             1.0.2             Reconcile succeeded   5d22h
tap-install   buildservice               buildservice.tanzu.vmware.com                  1.4.3             Reconcile succeeded   5d22h
tap-install   cartographer               cartographer.tanzu.vmware.com                  0.2.2             Reconcile succeeded   5d22h
tap-install   cert-manager               cert-manager.tanzu.vmware.com                  1.5.3+tap.1       Reconcile succeeded   5d22h
tap-install   cnrs                       cnrs.tanzu.vmware.com                          1.1.1             Reconcile succeeded   5d22h
tap-install   contour                    contour.tanzu.vmware.com                       1.18.2+tap.1      Reconcile succeeded   5d22h
tap-install   conventions-controller     controller.conventions.apps.tanzu.vmware.com   0.5.1             Reconcile succeeded   5d22h
tap-install   developer-conventions      developer-conventions.tanzu.vmware.com         0.5.0             Reconcile succeeded   5d22h
tap-install   fluxcd-source-controller   fluxcd.source.controller.tanzu.vmware.com      0.16.3            Reconcile succeeded   5d22h
tap-install   ootb-delivery-basic        ootb-delivery-basic.tanzu.vmware.com           0.6.1             Reconcile succeeded   5d22h
tap-install   ootb-supply-chain-basic    ootb-supply-chain-basic.tanzu.vmware.com       0.6.1             Reconcile succeeded   5d22h
tap-install   ootb-templates             ootb-templates.tanzu.vmware.com                0.6.1             Reconcile succeeded   5d22h
tap-install   service-bindings           service-bindings.labs.vmware.com               0.6.1             Reconcile succeeded   5d22h
tap-install   services-toolkit           services-toolkit.tanzu.vmware.com              0.5.1             Reconcile succeeded   5d22h
tap-install   source-controller          controller.source.apps.tanzu.vmware.com        0.2.1             Reconcile succeeded   5d22h
tap-install   spring-boot-conventions    spring-boot-conventions.tanzu.vmware.com       0.3.0             Reconcile succeeded   5d22h
tap-install   tap                        tap.tanzu.vmware.com                           1.0.2             Reconcile succeeded   5d22h
tap-install   tap-gui                    tap-gui.tanzu.vmware.com                       1.0.2             Reconcile succeeded   5d22h
tap-install   tap-telemetry              tap-telemetry.tanzu.vmware.com                 0.1.4             Reconcile succeeded   5d22h
tap-install   tekton-pipelines           tekton.tanzu.vmware.com                        0.30.1            Reconcile succeeded   5d22h
```

_Originally posted by @cmoulliard in https://github.com/vmware-tanzu/kubeapps/issues/4524#issuecomment-1083071986_

**Comment by absoludity** [comment]
The same improvement that was done in #4378 can be done for GetInstalledPackageSummaries.

**Comment by cmoulliard** [comment]
What is the status about this issue @absoludity ?

**Comment by absoludity** [comment]
> What is the status about this issue @absoludity ?

We've just added it to the current iteration, so hope to work on this this week or next.

**Comment by absoludity** [comment]
Fixed in the attached PR #4607 . I've only tested it with 10 installed packages, and it's around 1/6th of the response time. Feel free to try it out. There are more optimizations that could be done there, but this should be enough to make it feel much more responsive (and avoid timeouts).
